benches/eth_erc20.rs

```diff
@@ -0,0 +1,269 @@
+use aurora_engine::prelude::{Address, U256};
+use aurora_engine::transaction::EthTransaction;
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+use near_vm_logic::VMOutcome;
+use secp256k1::SecretKey;
+use std::path::{Path, PathBuf};
+
+// We don't use everything in `common`, but that's ok, other benchmarks do
+#[allow(dead_code)]
+mod common;
+mod solidity;
```

Comment:
I suggest using the following layout for benchmarks:

```
benches/
  main.rs
  eth_erc20.rs
  ...
```

where `main.rs` looks like this:

```
mod eth_erc20;
...

Stuff which is currently in common.rs
```

and with the following Cargo.toml:

```
[package]
...
autobenches = false

[[bench]]
name = "benches"
path = "benches/main.rs"
```

That way, you'll have only single integration binary compiled, which should save you a lot of time on linking, considering the fact that benchmarks are compiled in release mode. As a side-effect, you won't need `allow(deadcode)`. 

See https://github.com/near/nearcore/pull/4027 for details.

Reply:
How does this interact with the `criterion_group` and `criterion_main` macros I'm using at the bottom of each file? I assume I somehow want to say in `main.rs` that the benchmarks I want to run are contained in the various sub-modules.

Reply:
Good question! I haven't used criterion myself, but I think the following should work:

```rust
// benches/main.rs
mod eth_erc20;
mod eth_deploy_code;

criterion_group!(benches, eth_erc20::eth_erc20_benchmark, eth_deploy_code::deploy_code_benchmark);
criterion_main!(benches);

// benches/eth_erc20.rs

pub fn eth_erc20_benchmark(c: &mut Criterion) { ... }
```

Both group and main accept paths: https://docs.rs/criterion/0.3.4/criterion/macro.criterion_group.html

Alternatively, you can define a separate `criterion_group` in every submodule, and register all groups with criterion_main in main.rs, but it seems that you have one bench for submodule, so this isn't really neceesary.